frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/vms/NewVmModelBehavior.java

```diff
@@ -259,8 +263,13 @@ public void defaultHost_SelectedItemChanged() {
     @Override
     public void provisioning_SelectedItemChanged() {
         boolean provisioning = getModel().getProvisioning().getEntity();
-        getModel().getProvisioningThin_IsSelected().setEntity(!provisioning);
-        getModel().getProvisioningClone_IsSelected().setEntity(provisioning);
+        if (provisioning && isLatestTemplateSelected()) {
+            getModel().getProvisioningThin_IsSelected().setEntity(true);
+            getModel().getProvisioningClone_IsSelected().setEntity(false);
+        } else {
+            getModel().getProvisioningThin_IsSelected().setEntity(!provisioning);
+            getModel().getProvisioningClone_IsSelected().setEntity(provisioning);
```

Comment:
I've looked into the code and we use the template id, not the original template id

https://github.com/oVirt/ovirt-engine/blob/277acfcd95c21080d3251ce452068e422da41878/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/UpdateVmVersionCommand.java#L99

Reply:
yeah, I see it's also this way [in the stored procedure](https://github.com/oVirt/ovirt-engine/blob/98e2aa1cba2020839088e23c20aba5a1c507453e/packaging/dbscripts/vms_sp.sql#L2151-L2187) - and that's somehow debatable because the fact that I choose to clone the images, and there's no hard dependency anymore between the vm and the template (i.e., the template can be removed), doesn't mean that I'm not interested in having the VM updated when a new version is created. why do we need to include this change in this PR?

Reply:
reading the following part in the description: ```When creating a VM, the clone provisioning is selected by default. That does not make much sense for the latest template because the purpose of the latest template version is to be able to apply template's latest changes to the vm. This patch changes the selected provisioning to thin if the latest template is selected.```
this happens because the default vm type is now `server` so the disks are cloned, if we keep it as server but change disks provisioning to thin-provisioning - that would create a conflict (another argument why not to mix this change with the one that disables the stateless field for latest template versions)

Reply:
But what sense does it make to create a VM based on a latest Template when after creation you end up with latest Blank template? You could have used either directly the latest Blank template or a concrete version. So maybe we should filter the latest templates out of the dropbox if the provisioning is not thin instead of setting it by default? The same way as for the Stateless field?

Reply:
As we are not agreeing on the provisioning part, I'll remove it and keep just the stateless part of the patch,